server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -184,6 +185,14 @@
      */
     public static final Setting<Boolean> COMPRESS_SETTING = Setting.boolSetting("compress", true, Setting.Property.NodeScope);
 
+    /**
+     * When set to {@code true}, {@link #bestEffortConsistency} will be set to {@code true} and concurrent modifications of the repository
+     * contents will not result in the repository being marked as corrupted.
+     * Note: This setting is intended as a backwards compatibility solution for 7.x and will go away in 8.
+     */
+    public static final Setting<Boolean> ALLOW_CONCURRENT_MODIFICATION =
+        Setting.boolSetting("allow_concurrent_mod", false, Setting.Property.Deprecated, Setting.Property.Dynamic);
```

Comment:
Let's fully spell out the name, i.e. `allow_concurrent_modifications`

Reply:
Also, do we need to define NodeScope but not dynamic, like we have for COMPRESS_SETTING?

Reply:
> Also, do we need to define NodeScope but not dynamic, like we have for COMPRESS_SETTING?

Not really, all these attributes are just descriptive only as far as I can tell since these are not registered settings but and simply applied to the repository metadata. I'll remove the `Dynamic` part as it's pointless :)